src/util/image.cc

```diff
@@ -15,6 +15,17 @@ image::image(const char* image_filename) {
     std::cerr << "ERROR: Could not load image file '" << image_filename << "'.\n";
 }
 
+image::image(const char* image_filename, int bytes_per_pixel) : bytes_per_pixel{bytes_per_pixel} {
```

Comment:
Instead of creating an entirely new constructor, if the only difference is whether bytes_per_pixel takes on an assigned value or the default of 3, you could:
1. remove the default assignment from image.hh
2. make one constructor which takes in bytes_per_pixel as an optional param
3. set the default value of bytes_per_pixel = 3 for the param